atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -62,16 +63,21 @@ def run(self):
 
             base_image_with_registry.registry = self.parent_registry
 
-        pulled_base = self.tasker.pull_image(base_image_with_registry,
-                                             insecure=self.parent_registry_insecure)
-        if (base_image_with_registry.namespace != 'library' and
-                not self.tasker.image_exists(base_image_with_registry.to_str())):
+        try:
+            self.tasker.pull_image(base_image_with_registry,
+                                   insecure=self.parent_registry_insecure)
+
+        except RetryGeneratorException:
+            if base_image_with_registry.namespace == 'library':
+                raise
```

Comment:
One thing I just thought of: this will report a failure to pull e.g. "library/mything" whereas the developer might have "FROM mything". Should we make any attempt to raise the *original* exception when both pulls fail?

Reply:
which original exception ? you mean protocol & api error? and in case it didn't raise but was is_failed()?

1st it tries to pull image regardless if there is 'library'

if that fails, that mean that image isn't accessible, so this tries another pull with 'library', and if that works we will continue to work with that

Reply:
I mean that when a developer tries to build from a Dockerfile that has "FROM mything:1.0" but mything:1.0 is not present in the registry, we try again by pulling library/mything:1.0 (this "library/" bit is a left-over from the API v1 days).

So the error the developer will see is something like:
```
failed to pull image ("library/mything:1.0",): {"message": "unknown: Not Found")}
```

but a more useful message might be:
```
failed to pull image ("mything:1.0",): {"message": "unknown: Not Found")}
```

i.e. try pulling mything:1.0, but if the CommandResult.is_failed() then try pulling library/mything:1.0 and if it works then great. If that also fails, though, report the original CommandResult failure from pulling mything:1.0, not from pulling library/mything:1.0.

Without doing this, the error message says that library/mything:1.0 is not found even though this is expected to be the case (since only mything:1.0 is present), and developers would not be clear on what the problem actually is.

Make sense?

Reply:
yeah it does now